### PR TITLE
Ensure same cache if include and exclude are empty string / array.

### PR DESCRIPTION
**Pull request by spacedmonkey**
Trac ticket:  https://core.trac.wordpress.org/ticket/55352

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

File: src/wp-includes/class-wp-term-query.php [hunk 1]
```diff
@@ -457,6 +457,14 @@ public function get_terms() {
 				"tt.taxonomy IN ('" . implode( "', '", array_map( 'esc_sql', $taxonomies ) ) . "')";
 		}
 
+		if ( '' === $args['exclude'] ) {
+			$args['exclude'] = array();
+		}
+
+		if ( '' === $args['include'] ) {
+			$args['include'] = array();
+		}
```
Comment (hellofromtonya) [inline src/wp-includes/class-wp-term-query.php]: Could `exclude` be any empty or even a string that is only spaces? If yes, then I'd suggest the following:

```suggestion
		if ( empty( $args['exclude'] ) ) {
			$args['exclude'] = array();
		}

		if ( empty( $args['include'] ) ) {
			$args['include'] = array();
		}
```

**Comment by hellofromtonya** [comment]
Hey @spacedmonkey, can you share manual testing step-by-step instructions please including what code is needed to test it? This will help testers like me reproduce the issue and then validate the changes resolve it. Thanks!

**Review by peterwilsoncc — COMMENTED** [review]
My review is basically "what Tonya said" with some tests of those cases in the data provider too.

File: src/wp-includes/class-wp-term-query.php (hunk 1 above)
Comment (spacedmonkey) [reply src/wp-includes/class-wp-term-query.php]: Empty would also be trigger is 0 and '0', is this what we want here?

**Comment by spacedmonkey** [comment]
If you go the category list page in the wp-admin, then you will see a duplicate query, which is where I found the issue. See screenshot. 
Screenshot 2022-04-27 at 09 04 31

To test manually, I wrote this script. 

```
function test_init(){

	get_terms(array(
		'fields'           => 'all',
		'taxonomy'         => 'category',
		'exclude'          => '',
		'include'          => '',
	));
       get_terms( array(
		'taxonomy'         => 'category',
		'fields'           => 'all',
		'exclude'          => array(),
		'include'          => array()
      ));
}

add_action('init', 'test_init');
```

Then I used query monitor to see if a got duplicate queries.

**Comment by spacedmonkey** [comment]
>My review is basically "what Tonya said" with some tests of those cases in the data provider too.

There are extra example of this in the data provider? So this is tested.

File: src/wp-includes/class-wp-term-query.php (hunk 1 above)
Comment (peterwilsoncc) [reply src/wp-includes/class-wp-term-query.php]: > Empty would also be trigger is 0 and '0', is this what we want here?

Yes, that's good.

Currently `get_terms( [ 'include' => 0 ] );` runs the default query & returns `uncatagorized` on a fresh install. As there is never a term with the ID `0`, an emtpy array for `exclude` makes sense too.

**Comment by peterwilsoncc** [comment]
> There are extra example of this in the data provider? So this is tested.

Sorry, I was unclear -- I was thinking of `0` and maybe even `'0'` to make sure the result is as expected.

**Comment by peterwilsoncc** [comment]
I pushed up a switch to `empty()` and added a `array [?] vs no [?]` and `array [?] vs zero [?]` test for both include and exclude.

**Comment by spacedmonkey** [comment]
> I pushed up a switch to empty() and added a array [?] vs no [?] and array [?] vs zero [?] test for both include and exclude.

I am okay with that. Should we get this committed?

**Review by peterwilsoncc — APPROVED** [review]
LGTM. 

Did some manual testing this afternoon and it was working as expected

**Comment by spacedmonkey** [comment]
Committed

**Pull request closed without merging**